src/query/pipeline/transforms/src/processors/transforms/transform_sort_merge.rs

```diff
@@ -89,6 +91,52 @@ where
             _r: PhantomData,
         })
     }
+
+    fn adjust(
```

Comment:
It's better to abstract a struct for the loser tree.

Reply:
OK, I will implement this tonight and, in the process, conduct some testing for the 'order by' feature to identify any areas that may require optimization in the implementation.

Reply:
We also need unit tests for the loser tree.